### DISCUSSION TITLE
Why does mini.surround behave like this?

### DISCUSSION THREAD

**Discussion by limaceous-bushwhacker** [nvim-mini/mini.nvim#661] [Q&A] [answered]
```
  {
    "echasnovski/mini.surround",
    opts = {
      search_method = "cover_or_next",
    },
  },
```

If I do `sd` on L2 (opts), it'd then delete quote from above and below L2 instead of jumping down and deleting those. Seems a bit weird. This doesn't happen on tpope's surround module.

**Comment by echasnovski** [comment, marked as answer]
Because [quotes in 'mini.surround' are not balanced](https://github.com/echasnovski/mini.nvim/blob/910e49d2d45e9d32c815d16cce6cea7c3b43cecf/doc/mini-surround.txt#L121) by default. I don't *quite* remember the actual reason why this is the default, but it was deliberate (there is explicit test for that). My current reasoning is that all three quotes are not balanced pairs (they use same open and close characters) and I can see them closely to any other default surrounding (which is not balanced). Not very strong reason, I admit, but it is what it is now (as breaking change is bad).

Besides, you can modify them to be balanced like this:

```lua
require('mini.surround').setup({
  custom_surroundings = {
    ['"'] = { input = { '%b""', '^.().*().$' } },
  },
  search_method = 'cover_or_next',
})
```

**Reply by limaceous-bushwhacker** [reply]
Thanks for the workaround. Been using mini.surround for months now but this always had me scratching my head.
